extensions/frontend-tiles/src/GraphicsProvider/GraphicRepresentationProvider.ts

```diff
@@ -0,0 +1,255 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, Logger} from "@itwin/core-bentley";
+import { loggerCategory} from "../LoggerCategory";
+
+/** The format of the Graphic Representation
+ * @beta
+ */
+export enum GraphicRepresentationFormat {
+  IMDL = "IMDL",
+  Tiles3D = "3DTILES",
+}
+
+/** Graphic representations are generated from Data Sources.
+ * The status of a Graphic Representation indicates the progress of that generation process.
+ * @beta
+ */
+export enum GraphicRepresentationStatus {
+  InProgress = "In progress",
+  Complete = "Complete",
+  NotStarted = "Not started",
+  Failed = "Failed",
+}
+
+/**
+ * Represents a data source for a graphic representation.
+ * A data source is usually higher-fidelity and contains more information, but may not be as well suited for visualization
+ * as a graphic representation.
+ * @beta
+ */
+export interface DataSource {
+  /** The iTwinId associated with th data source */
+  iTwinId: string;
+  /** The id of the data source */
+  id: string;
+  /** The change id of the data source */
+  changeId?: string;
+  /** The type of the data source */
+  type: string;
+}
+
+/** represents a visual representation of data from a data source.
+ * This could be a 3D model, a map, or any other kind of graphical data.
+ * @see [[queryGraphicsDataSources]].
+ * @beta
+ */
+export interface GraphicRepresentation {
+  /** The displayName of the Graphic Representation */
+  displayName: string;
+  /** The Id of the Graphic Representation */
+  representationId: string;
+  /** The status of the query from the Graphics Provider for the Graphic Representation */
+  status: GraphicRepresentationStatus;
+  /** The format of the Graphic Representation*/
+  format: GraphicRepresentationFormat;
+  /** The url of the Graphic Representation */
+  url?: string;
+  /** The data source that the representation originates from.
+  * For example, a GraphicRepresentation in the 3D Tiles format might have a dataSource that is a specific iModel changeset.
```

Comment:
This is an example of a documentation comment that provides useful information. Revisit your other doc comments to make them do the same.

Reply:
Docs have been rewritten to be more useful.